### PR TITLE
Feature/add description to feature toggles

### LINKED ISSUE #44: Ability to add a description to feature toggles
When using the feature toggles it might be useful to have the ability to add metadata to a feature toggle to clarify the definition for the end user.

```yaml
nusje2000_feature_toggle:
    environment:
        features:
            feature_toggle_1: true
            feature_toggle_2: false
            feature_toggle_3:
                description: 'For human. By human.'
                value: false
```

### PR DESCRIPTION
**Pull request by KevinVanSonsbeek**
## Warning! this PR is BC breaking due to changes to the Feature interface
Fixes #44 

### todo:
- [x] Add methods to the Feature interface
- [x] Add description to the Create and Update controllers
- [x] Make the View controller return the description
- [x] Add description to the FeatureMapper 
- [x] Allow  proposed config options to be succesfully parsed

Proposed config options:
```yaml
nusje2000_feature_toggle:
    environment:
        features:
            feature_toggle_1: true
            feature_toggle_2: false
            feature_toggle_3:
                description: 'For human. By human.'
                value: false
```

### CODE REVIEW

File: src/DependencyInjection/Nusje2000FeatureToggleExtension.php [hunk 1]
```diff
@@ -105,7 +108,10 @@ private function configureLogger(ContainerBuilder $container, $loggerConfig): vo
      *     enabled: bool,
      *     name: string,
      *     hosts: list<string>,
-     *     features: array<array-key, bool>,
+     *     features: array<array-key, array{
+     *          value: bool,
```
Comment (nusje2000) [inline src/DependencyInjection/Nusje2000FeatureToggleExtension.php:112]: I would suggest using the name "state"

File: tests/Unit/Controller/Host/Feature/UpdateControllerTest.php [hunk 2]
```diff
@@ -63,4 +63,36 @@ public function testInvokeWithMissingState(): void
 
         $controller($request, 'environment', 'feature');
     }
+
```
Comment (nusje2000) [inline tests/Unit/Controller/Host/Feature/UpdateControllerTest.php:66]: Can you add a `testInvokeWithDescriptionOnly`?

**Review by nusje2000 — CHANGES_REQUESTED** [review]

File: src/Controller/Host/Feature/UpdateController.php [hunk 3]
```diff
@@ -58,4 +59,19 @@ private function updateFeature(Feature $feature, array $json): void
 
         $feature->disable();
     }
+
+    /**
+     * @param array<mixed> $json
+     */
+    private function updateFeatureDescription(Feature $feature, array $json): void
+    {
+        $description = $json['description'] ?? null;
```
[the remark this answers is not in the document]
Comment (KevinVanSonsbeek) [reply src/Controller/Host/Feature/UpdateController.php:68]: What would be more preferable behavior, checking if the key exists, if it does, check the type, and update it if it's not the same? Or explicitly pass false to description to clear it?

[the remark this answers is not in the document]
Comment (nusje2000) [reply src/Controller/Host/Feature/UpdateController.php:68]: I would suggest using an array_key_exists

File: src/DependencyInjection/Nusje2000FeatureToggleExtension.php (hunk 1 above)
Comment (nusje2000) [reply src/DependencyInjection/Nusje2000FeatureToggleExtension.php:112]: That would be a better option indeed 👍

File: tests/Unit/Controller/Host/Feature/UpdateControllerTest.php (hunk 2 above)
Comment (KevinVanSonsbeek) [reply tests/Unit/Controller/Host/Feature/UpdateControllerTest.php]: I've refactored the test to be based on a data provider, so more test cases can be added at ease.

File: src/DependencyInjection/Nusje2000FeatureToggleExtension.php (hunk 1 above)
Comment (KevinVanSonsbeek) [reply src/DependencyInjection/Nusje2000FeatureToggleExtension.php]: The option has been renamed from "value" to "enabled"

**Pull request merged**